test/internalTest.js

```diff
@@ -153,29 +153,38 @@ mineflayer.testedVersions.forEach((supportedVersion, i) => {
     })
     it('blockAt', (done) => {
       const pos = vec3(1, 65, 1)
-      const goldId = 41
+      const goldId = mcData.blocksByName.gold_block.id
       bot.on('chunkColumnLoad', (columnPoint) => {
         assert.strictEqual(columnPoint.x, 0)
         assert.strictEqual(columnPoint.z, 0)
         assert.strictEqual(bot.blockAt(pos).type, goldId)
         done()
       })
       server.on('login', (client) => {
-        client.write('login', {
+        const loginPacket = {
           entityId: 0,
           levelType: 'fogetaboutit',
-          hashedSeed: [0, 0],
           gameMode: 0,
           previousGameMode: 255,
           worldNames: ['minecraft:overworld'],
-          dimensionCodec: { name: '', type: 'compound', value: { dimension: { type: 'list', value: { type: 'compound', value: [w] } } } },
-          dimension: bot.supportFeature('dimensionIsAnInt') ? 0 : 'minecraft:overworld',
+          dimensionCodec: (version.version >= 735 ? mcData.loginPacket.dimension : { name: '', type: 'compound', value: { dimension: { type: 'list', value: { type: 'compound', value: [w] } } } }),
+          dimension: (version.version >= 735 ? mcData.loginPacket.dimension : 0),
```

Comment:
you can just use mcData.loginPacket instead of the big thing using w
it would be much better to put this version.version >= 735 inside support feature function (changing how it works)